### PR TITLE
Matt/room transition stats

### PR DESCRIPTION
**Pull request by ChatBotMatt**

### CODE REVIEW

File: main_scenes/game_scene.gd [hunk 1]
```diff
@@ -55,8 +57,9 @@ func _process(_delta):
 func _get_event() -> String:
 	if GlobalVariables.DungeonSequence.size():
 		return GlobalVariables.DungeonSequence.pop_front()
-	return "healing_room"
-	
+
+	return "healing_room" # TODO Code fails if we don't return anything
```
Comment (RossBrunton) [inline main_scenes/game_scene.gd:61]: This should maybe be an assert(false) then.

File: main_scenes/game_scene.gd [hunk 2]
```diff
@@ -126,6 +127,7 @@ func _finish_moving_to_next_room():
 func run_event(event_id: String):
 	assert(event_state == EventState.IDLE)
 	assert(current_event == null)
+	print(event_id)
```
Comment (RossBrunton) [inline main_scenes/game_scene.gd:130]: Debug statement left in code.

File: globals/Party.gd [hunk 3]
```diff
@@ -144,13 +144,13 @@ func MotivateParty(heal: int, bonus: Variant = null):
 			hero.mv += with_bonus
 			hero.mv = min(hero.mv, hero.max_mv)
 
-func GrantExp(exp: int, bonus: Variant = null):
+func GrantExp(exp_value: int, bonus: Variant = null):
```
Comment (RossBrunton) [inline globals/Party.gd:147]: Change not relevant to MR.

File: main_scenes/game_scene.gd [hunk 4]
```diff
@@ -134,7 +136,7 @@ func run_event(event_id: String):
 	$GUI/GoButton.visible = true
 
 # Send text to the dialogue box, and play it
-func _send_text(text: String, name: String = "MinionNo1", use_right_sprite: bool = false):
+func _send_text(text: String, text_name: String = "MinionNo1", use_right_sprite: bool = false):
```
Comment (RossBrunton) [inline main_scenes/game_scene.gd:139]: Change not relevant to MR.

File: main_scenes/game_scene.gd (hunk 1 above)
Comment (ChatBotMatt) [reply main_scenes/game_scene.gd:61]: @Cmckenzie01

File: main_scenes/game_scene.gd (hunk 2 above)
Comment (ChatBotMatt) [reply main_scenes/game_scene.gd:130]: Good catch, thanks

File: globals/Party.gd (hunk 3 above)
Comment (ChatBotMatt) [reply globals/Party.gd:147]: It's to stop it shadowing a builtin, so helps stop future bugs. Admitttedly maybe not relevant at this point.

File: main_scenes/game_scene.gd (hunk 4 above)
Comment (ChatBotMatt) [reply main_scenes/game_scene.gd:139]: It's to stop it shadowing a builtin, so helps stop future bugs. Admitttedly maybe not relevant at this point.

**Pull request merged**